pkg/resource/table/hooks_local_secondary_indexes.go

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package table
+
+import (
+	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
+	ackutil "github.com/aws-controllers-k8s/runtime/pkg/util"
+
+	"github.com/aws-controllers-k8s/dynamodb-controller/apis/v1alpha1"
+)
+
+func computeLocalSecondaryIndexDelta(
```

Comment:
Are local secondary indexes ever updated?

Reply:
Local secondary indexes can only be set during the CREATE op. They are immutable. `computeLocalSecondaryIndexDelta` is used to set a terminal condition if users try to modify them